src/infrastructure/api/meal-provider/neis_api/neis-api.ts

```diff
@@ -4,67 +4,102 @@ import Neis from "@my-school.info/neis-api";
 import { IMealInfoRow, ISchoolInfoRow } from "@my-school.info/neis-api/dist/interpaces/response";
 import { Meal } from "@src/meal/domain/meal";
 import { MealProvider } from "../meal-provider";
+import { MealRepositoryImpl } from "@src/infrastructure/repository/meal.repository.impl";
+import { MealRepository } from "@src/infrastructure/repository/meal.repository";
 
 @Injectable()
 export class NeisApi implements MealProvider {
     constructor(
         @Inject(NeisApiClient)
         private readonly neisApiClient: Neis,
         @Inject(SchoolInfo)
-        private readonly schoolInfo: ISchoolInfoRow
+        private readonly schoolInfo: ISchoolInfoRow,
+        @Inject(MealRepositoryImpl)
+        private readonly mealRepository: MealRepository
     ) {}
 
+    async saveMonthlyMeal() {
+        const currentDate = new Date();
+        const lastDateOfMonth = this.getLastDateOfMonth(currentDate.getFullYear(), currentDate.getMonth());
+        const firstDayOfMonth = new Date(currentDate.getFullYear(), currentDate.getMonth(), 1);
+        for (let i = 1; i < lastDateOfMonth; i++) {
+            firstDayOfMonth.setDate(firstDayOfMonth.getDate() + 1);
+            const rows: IMealInfoRow[] = await this.getNeisMealInfo(firstDayOfMonth);
+
+            if (rows == undefined) {
+                continue;
+            }
+
+            const meal: Meal = new Meal();
+
+            rows.forEach((row: IMealInfoRow) => {
+                const type: number = +row.MMEAL_SC_CODE;
+                const menu: string = this.replaceMenu(row);
+                this.setMealType(meal, type, menu);
+            });
+
+            await this.mealRepository.saveMeal(meal, firstDayOfMonth);
+        }
+    }
+
+    replaceMenu(row: IMealInfoRow): string {
+        return row.DDISH_NM.replace(/([0-9]{1,5}\.)+/g, "")
+            .replace(/\(\)/g, "")
+            .replace(/\s/g, "")
+            .replace(/\./g, "")
+            .replace(/\(\d+\)/g, "")
+            .split("<br/>")
+            .concat(row.CAL_INFO)
+            .join("||");
+    }

```

Comment:
나누는게 보기 더 편하면 안 바꿔도 되긴 하지만 이렇게 합치는 것도 괜찮을 것 같습니당

```suggestion
    replaceMenu(row: IMealInfoRow): string {
        return row.DDISH_NM.replace(/([0-9]{1,5}\.)+/g, "")
            .replace(/\(\)|\s|\(\d+\)|\./g, "")
            .split("<br/>")
            .concat(row.CAL_INFO)
            .join("||");
    }

```